### PR TITLE
Moved ConfirmProvider into AppLayout

### PR DESCRIPTION
**Pull request by nikitharani**
Contributes to [TD-1667](https://sce.myjetbrains.com/youtrack/issue/TD-1667/Fix-Images-for-MM-in-Dark-and-Light-mode)

## Changes

- Migrated AppLayout, AppLauncher, AppHeader, Sidebar, user menu, and NowrapTypography Component to TS
- imported and added the ConfirmProvider component in AppLayout to avoid importing ConfirmProvider in individual apps.
- removed divider from SummaryCard

Maybe a short description of what is not included and will come in future work where appropriate.

- Haven't migrated SummaryCard to Ts this will be done in future sprints.

## Testing notes
- You should run the npm test to confirm that the tests work. 

- You should run npm run build and check that the dist/ folder contains .d.ts files containing the type definitions

## Author checklist before assigning a reviewer

- [x] Reviewed my own code-diff.
- [x] Assigned to me.
- [x] Relevant tags added.
- [x] Appropriate tests have been added.
- [x] Lint and test workflows pass.

### CODE REVIEW

File: src/AppHeader/AppHeader.types.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+   *
+   * _newMode_: The new color mode that has been selected
+   */
+  onModeChange: (newMode: "light" | "dark") => void;
+  /**
+   * Name of currently logged in user.
+   */
+  username: string;
+  /**
+   * A String of the href URL for the Link of the VIRTO Logo, default is null (link disabled)
```
Comment (mattcorner) [inline src/AppHeader/AppHeader.types.ts:51]: You've changed this to not use null for some reason, but haven't updated the doc. Be consistent.

File: src/UserMenu/UserMenu.types.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+export interface UserMenuProps {
+  /**
+   * Callback fired when the user clicks on "Change password".
+   *
+   * **Signature**
+   * ```
+   * function(event: object) => void
+   * ```
```
[the remark this answers is not in the document]
Comment (mattcorner) [reply src/UserMenu/UserMenu.types.ts:8]: image

As shown here, the typescript signature is added below our documentation.

**Comment by mattcorner** [comment]
@nikitharani You've made UI changes here, but not included any UI screenshots in the PR description for @Sowbhagya-ipg to review. Please make sure you do this.

@Sowbhagya-ipg Here is the change that removes the divider in the SummaryCard. Does this match your expectation?

image

**Review by mattcorner — CHANGES_REQUESTED** [review]
Some typescript based stuff to fix.

**Comment by Sowbhagya-ipg** [comment]
> @nikitharani You've made UI changes here, but not included any UI screenshots in the PR description for @Sowbhagya-ipg to review. Please make sure you do this.
> 
> @Sowbhagya-ipg Here is the change that removes the divider in the SummaryCard. Does this match your expectation?
> 
> image

1. @mattcorner @nikitharani View all button is not needed at all.
2. The Left side text should be of secondary color
3. Removing the divider is alright here, But the extra empty space is not needed. The height should adjust based on the content

**Comment by Sowbhagya-ipg** [comment]
> > > 
> > 
> > 
> > > > > @nikitharani You've made UI changes here, but not included any UI screenshots in the PR description for @Sowbhagya-ipg to review. Please make sure you do this.
> > > > > @Sowbhagya-ipg Here is the change that removes the divider in the SummaryCard. Does this match your expectation?
> > > > > image
> > > > 
> > > > 
> > > > 
> > > > 1. @mattcorner @nikitharani View all button is not needed at all.
> > > > 2. The Left side text should be of secondary color
> > > > 3. Removing the divider is alright here, But the extra empty space is not needed. The height should adjust based on the content
> > > 
> > > 
> > > 
> > > 1. Is just an example of usage. We can update the story if you don't ever see a use case for >1 button.
> > > 2. The extra space is a valid use case I believe. e.g. in VIRTO.MODEL different summary cards show different numbers of metadata according to Figma.
> > 
> > 
> > The extra space will be adjusted in apps by adding/adjusting the height of the card based on the number of metadata. Here I can adjust height inside the story to match with Figma
> 
> Updated CardComponent image

This is good.

File: src/Card/SummaryCard/SummaryCard.stories.js [hunk 3]
```diff
@@ -16,6 +16,7 @@ import {
 } from "@mui/material";
 import { Delete, Edit } from "@mui/icons-material";
 
+// import { NoWrapTypography } from "../../NoWrapTypography";
```
Comment (mattcorner) [inline src/Card/SummaryCard/SummaryCard.stories.js:19]: Remove this line.

**Review by mattcorner — CHANGES_REQUESTED** [review]

File: src/Sidebar/SidebarItem/SidebarItem.types.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+   */
+  disabled?: boolean;
+  /**
+   * Icon to display alongside text.
+   */
+  icon: React.ReactElement;
+  /**
+   * Custom style to apply to the icon
+   */
+  iconStyle?: Object;
```
[the remark this answers is not in the document]
Comment (nikitharani) [reply src/Sidebar/SidebarItem/SidebarItem.types.ts:27]: added CssProperties in commit [f6ef24c](https://github.com/IPG-Automotive-UK/react-ui/pull/579/commits/f6ef24cd2b3256e70f2d48e77d0191c8ea0dc852)

[the remark this answers is not in the document]
Comment (nikitharani) [reply src/Sidebar/SidebarItem/SidebarItem.types.ts]: ignore the previous comment resolved now.

**Review by mattcorner — APPROVED** [review]

**Pull request merged**